src/query/ee/src/storages/fuse/operations/vacuum_drop_tables.rs

```diff
@@ -15,120 +15,143 @@
-        }
-    };
-
-    info!(
-        "vacuum drop table {:?} dir {:?}, cost:{} sec",
-        table_info.name,
-        dir,
-        start.elapsed().as_secs()
-    );
-    ret
+        info!(
+            "vacuum drop table {:?} dir {:?}, cost:{} sec",
+            table_info.name,
+            dir,
+            start.elapsed().as_secs()
+        );
+    }
+    Ok(if dry_run_limit.is_some() {
+        Some(list_files)
+    } else {
+        None
+    })
 }
 
 #[async_backtrace::framed]
 pub async fn do_vacuum_drop_tables(
+    threads_nums: usize,
     tables: Vec<Arc<dyn Table>>,
     dry_run_limit: Option<usize>,
 ) -> Result<Option<Vec<VacuumDropFileInfo>>> {
     let start = Instant::now();
     let tables_len = tables.len();
     info!("do_vacuum_drop_tables {} tables", tables_len);
-    let mut list_files = Vec::new();
-    let mut left_limit = dry_run_limit;
-    for table in tables {
-        // only operate fuse table
-        let ret = if let Ok(fuse_table) = FuseTable::try_from_table(table.as_ref()) {
-            let table_info = table.get_table_info();
-            let operator = fuse_table.get_operator_ref();
-            do_vacuum_drop_table(table_info, operator, left_limit).await?
+
+    let batch_size = (tables_len / threads_nums).min(50).max(1);
+
+    let result = if batch_size >= tables.len() {
+        do_vacuum_drop_table(tables, dry_run_limit).await?
+    } else {
+        let mut chunks = tables.chunks(batch_size);
+        let thread_limit = if let Some(dry_run_limit) = dry_run_limit {
+            Some((dry_run_limit / threads_nums).min(dry_run_limit).max(1))
         } else {
-            info!(
-                "ignore table {}, which is not of FUSE engine. Table engine {}",
-                table.get_table_info().name,
-                table.engine()
-            );
-            continue;
+            None
         };
-        if let Some(ret) = ret {
-            list_files.extend(ret);
-            if list_files.len() >= dry_run_limit.unwrap() {
-                info!(
-                    "do_vacuum_drop_tables {} tables, cost:{} sec",
-                    tables_len,
-                    start.elapsed().as_secs()
-                );
-                return Ok(Some(list_files));
-            } else {
-                left_limit = Some(dry_run_limit.unwrap() - list_files.len());
+        let tasks = std::iter::from_fn(move || {
+            chunks
+                .next()
+                .map(|tables| do_vacuum_drop_table(tables.to_vec(), thread_limit))
+        });
+
+        let result = execute_futures_in_parallel(
+            tasks,
+            threads_nums,
+            threads_nums * 2,
+            "batch-vacuum-drop-tables-worker".to_owned(),
+        )
+        .await?;
+        if dry_run_limit.is_some() {
+            let mut ret_files = vec![];
+            for file in result {
+                if let Ok(Some(files)) = file {
+                    ret_files.extend(files);
+                }
```

Comment:
This ignores any exceptions that might occur during the executions of `do_vacuum_drop_table`, which is inconsistent with the previous behavior, should we consider throwing the exceptions?

Reply:
```                
// return error if any errors happens during `do_vacuum_drop_table`
                if let Err(e) = file {
                    return Err(e);
                } else if let Ok(Some(files)) = file {
                    ret_files.extend(files);
                }
            }
```